### PR TITLE
Add legacy ignoreNulls option to array_agg

### PR DESCRIPTION
**Pull request by pranjalssh**
Addresses https://github.com/facebookincubator/velox/issues/5843

### CODE REVIEW

**Review by mbasmanova — COMMENTED** [review]
@pranjalssh I wonder if it might be better to introduce a configuration property to control this behavior. 

For example, there are a few configs that alter the behavior of Spark functions: https://facebookincubator.github.io/velox/configs.html#spark-specific-configuration

**Comment by pranjalssh** [comment]
@mbasmanova Sounds good, updated PR to use a velox config.

File: velox/core/QueryConfig.h [hunk 1]
```diff
@@ -188,6 +188,9 @@ class QueryConfig {
   static constexpr const char* kSpillableReservationGrowthPct =
       "spillable_reservation_growth_pct";
 
+  /// If true, array_agg() aggregation function will ignore nulls in the input.
+  static constexpr const char* kArrayAggIgnoreNulls = "array_agg_ignore_nulls";
```
Comment (mbasmanova) [inline velox/core/QueryConfig.h:192]: It would be nice to document this config in velox/docs/configs.rst

Would it make sense to add presto. prefix to the name to clarify that this config applies to Presto function? Similar to spark.xxx configs?

presto.array_agg.ignore_nulls

?

File: velox/functions/prestosql/aggregates/ArrayAggAggregate.cpp [hunk 2]
```diff
@@ -55,11 +56,22 @@ class ArrayAggAggregate : public exec::Aggregate {
...
     BufferPtr nulls = allocateNulls(numRows, pool);
+    auto mutableNulls = nulls->asMutable<uint64_t>();
     memcpy(
         nulls->asMutable<uint64_t>(),
         rows.asRange().bits(),
         bits::nbytes(numRows));
 
+    auto loadedElements = BaseVector::loadedVectorShared(elements);
+
+    if (ignoreNulls_) {
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/ArrayAggAggregate.cpp:67]: nit: perhaps add && loadedElements->mayHaveNulls()

File: velox/functions/prestosql/aggregates/tests/ArrayAggTest.cpp [hunk 3]
```diff
@@ -32,13 +34,25 @@ class ArrayAggTest : public AggregationTestBase {
  protected:
   void SetUp() override {
     AggregationTestBase::SetUp();
+  }
 
+  static void SetUpTestCase() {
+    functions::prestosql::registerAllScalarFunctions();
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/tests/ArrayAggTest.cpp:40]: Why is this needed?

File: velox/functions/prestosql/aggregates/tests/ArrayAggTest.cpp [hunk 4]
```diff
@@ -64,24 +78,29 @@ TEST_F(ArrayAggTest, groupBy) {
...
+        fmt::format(
+            "SELECT c0, array_agg(a) {}, max(c0) FROM tmp GROUP BY c0", filter),
+        makeConfig(ignoreNulls));
   };
 
-  testFunction("array_agg");
-  testFunction("simple_array_agg");
+  testFunction("array_agg", true);
+  testFunction("array_agg", false);
+  testFunction("simple_array_agg", false);
```
Comment (mbasmanova) [inline velox/functions/prestosql/aggregates/tests/ArrayAggTest.cpp:103]: @kagamiori Wei, looks like we need to update simple_array_agg as well.

**Review by mbasmanova — APPROVED** [review]
@pranjalssh Looks good. Thank you for adding this functionality. A few minor comments.

File: velox/core/QueryConfig.h (hunk 1 above)
Comment (mbasmanova) [inline velox/core/QueryConfig.h:192]: Let's also update documentation for array_agg itself to explain that its behavior depends on a config.

File: velox/functions/prestosql/aggregates/tests/ArrayAggTest.cpp (hunk 3 above)
Comment (pranjalssh) [reply velox/functions/prestosql/aggregates/tests/ArrayAggTest.cpp]: cleaned up code here.
